app/src/main/java/com/saucelabs/mydemoapp/android/view/fragments/ProductCatalogFragment.java

```diff
@@ -35,6 +35,15 @@ public class ProductCatalogFragment extends BaseFragment implements View.OnClick
     ProductsAdapter adapter;
     ProductCatalogViewModel viewModel;
 
+    public static boolean isRunningEspressoTest() {
```

Comment:
I would like to understand if this check is actually needed. I mean, why can't we just enable nested scrolling `android:nestedScrollingEnabled="false"`

Reply:
This is a better approach indeed. If it works, please ignore my alternative solution (accessing activity reference in test code) above.

Reply:
> I would like to understand if this check is actually needed. I mean, why can't we just enable nested scrolling `android:nestedScrollingEnabled="true"`

As I mentioned in description, scrolling down time to time ends up with scrolling up which breaks taking full page screenshot in our tests. The issue is fixed when I turn off `nestedScrollingEnabled`

Reply:
Thanks!

Reply:
Then please create a function in the fragment that does this:

```
binding.productRV.setNestedScrollingEnabled(true);
```

and call that function explicitly in the test class so that you can safely remove `isRunningEspressoTest()` from the fragment entirely.